src/main/java/seedu/address/model/util/SampleDataUtil.java

```diff
@@ -86,28 +109,41 @@ public static User getSampleUser() {
      * @param personData String representing the person data in the text file.
      */
     private static Person getSamplePerson(String personData) {
-        String[] personDataAsArray = personData.split("\\|");
-        Name name = new Name(personDataAsArray[0]);
-        Phone phone = new Phone(personDataAsArray[1]);
-        Email email = new Email(personDataAsArray[2]);
-        Address address = new Address(personDataAsArray[3]);
-        TelegramHandle telegramHandle = new TelegramHandle(personDataAsArray[4]);
-        Set<GroupTag> groupTagSet = getGroupTagSet(personDataAsArray[5].split(" "));
-        Set<ModuleTag> moduleTagSet = getModuleTagSet(personDataAsArray[6].split(" "));
+        List<String> personDataList = Stream.of(personData.split("\\|"))
+                .map(String::trim)
+                .limit(7) // Maximum of 7 fields
+                .collect(Collectors.toList());
+
+        if (personDataList.size() != 7) {
+            return null;
+        }
+
+        Name name = new Name(personDataList.get(0));
+        Phone phone = new Phone(personDataList.get(1));
+        Email email = new Email(personDataList.get(2));
+        Address address = new Address(personDataList.get(3));
+        TelegramHandle telegramHandle = new TelegramHandle(personDataList.get(4));
+        Set<GroupTag> groupTagSet = getGroupTagSetFromUnsplitted(personDataList.get(5));
+        Set<ModuleTag> moduleTagSet = getModuleTagSetFromUnsplitted(personDataList.get(6));
+
         return new Person(name, phone, email, address,
                 telegramHandle, groupTagSet, moduleTagSet);
     }
 
     /**
      * Returns a sample array of Persons.
      */
-    public static Set<Person> getSamplePersons() throws FileNotFoundException {
+    public static List<Person> getSamplePersons() throws FileNotFoundException {
```

Comment:
There's no shuffling a set so here it is.

Reply:
LGTM!